### ISSUE TITLE
How to ignore a specific regions in object detection?

### ISSUE THREAD

**Issue by NOTGOOOOD** [ultralytics/ultralytics#21079]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, dear developer:

Some datasets contains `ignore regions` bounding boxes annotation, but I haven't found how such annotations can be used in yolo.

I have two ideas for doing this:

1. Fill the 'ignore' area directly with a random color, so that it can be safely considered as the 'background'.

2. Create a new class for the 'ignore region' called '_ignore_', and rewrite the training logic to prevent features in the ignore regions to participate back-propagation.

I'm wondering if some similar operation is already included in the ultralytics library. If not, can you provide some suggestions?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NOTGOOOOD, thank you for reaching out and for your interest in Ultralytics 🚀! For your question about handling ignore regions in object detection datasets, we recommend checking the [Docs](https://docs.ultralytics.com/), which provide many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples and might already address similar use-cases.

Since this is a custom training ❓ question, please share as much detail as possible—such as dataset samples, annotation formats, and any relevant training logs—so we can better understand your setup. If you have a specific implementation or a small [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) demonstrating your approach, that would be very helpful!

This is an automated response 🤖. An Ultralytics engineer will review your issue and assist you further soon.

Join our community for more support and real-time discussion! Hop onto [Discord](https://discord.com/invite/ultralytics) 🎧, share ideas on [Discourse](https://community.ultralytics.com/), or explore threads on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

First, please make sure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
There's no internal support for this. You would have to modify `loss.py` to add gradient masking:

https://github.com/ultralytics/ultralytics/blob/16a20ac09bed1f513841d073a77fbb71eaf10483/ultralytics/utils/loss.py

**Comment by NOTGOOOOD** [comment]
Thanks, I will try

**Comment by glenn-jocher** [comment]
You're welcome! Feel free to open a new issue if you encounter any challenges while implementing the gradient masking approach in the loss function. Good luck with your implementation! 👍

**Comment by lyzln** [comment]
Hello，I meet the same question，please ask how to deal to it?

**Comment by glenn-jocher** [comment]
Not built-in; the practical route is to pass your ignore boxes through the dataloader (e.g., batch["ignore"]) and in ultralytics/utils/loss.py mask anchors whose centers fall inside them—after make_anchors/assigner compute a boolean mask from anchor_points vs ignore boxes, then set target_scores[mask] = 0 and fg_mask[mask] = False so BCE/IoU/DFL skip those anchors; see the tensors and flow in the reference for ultralytics/utils/loss.py.
